### PR TITLE
Match xlMain

### PR DESCRIPTION
**Pull request by cadmic**
and simulatorGetArgument, and shuffle some stuff around

### CODE REVIEW

File: include/simGCN.h [hunk 1]
```diff
@@ -104,11 +113,14 @@ extern void* gpSound;
 extern System* gpSystem;
 
 void simulatorUnpackTexPalette(__anon_0xDB69* pal);
+s32 simulatorDVDOpen(char* szNameFile, struct DVDFileInfo* pFileInfo);
+s32 simulatorDVDRead(struct DVDFileInfo* pFileInfo, void* anData, s32 nSizeRead, s32 nOffset,
```
Comment (Yanis42) [inline include/simGCN.h]: is ``struct`` necessary here? should be a sdk thing right?

File: include/simGCN.h [hunk 2]
```diff
@@ -104,11 +113,14 @@ extern void* gpSound;
...
+                     void (*callback)(s32, struct DVDFileInfo*));
```
Comment (Yanis42) [inline include/simGCN.h]: the callback here should be ``DVDCallback`` from dvd.h btw

File: src/simGCN.c [hunk 3]
```diff
@@ -343,6 +355,222 @@ const f32 D_80135D6C = 120.0;
...
+s32 xlMain() {
+    // Local variables
+    GXColor color; // r1+0x3C
+    SystemMode eMode; // r1+0x38
+    s32 nSize0; // r1+0x34
+    s32 nSize1; // r1+0x30
+    s32 iName; // r5
+    char* szNameROM; // r1+0x2C
+    char acNameROM[32]; // r1+0xC
+    // s32 rumbleYes; // r1+0x8
```
Comment (Yanis42) [inline src/simGCN.c:389]: should we keep these comments? I usually remove them once I'm done matching a function

**Review by Yanis42 — COMMENTED** [review]
small things

File: include/simGCN.h (hunk 1 above)
Comment (Yanis42) [reply include/simGCN.h]: nvm you changed that before I sent the review lol

File: src/simGCN.c (hunk 3 above)
Comment (cadmic) [reply src/simGCN.c]: I was too excited and forgot to clean up

Comment (cadmic) [reply src/simGCN.c]: I kept `// s32 rumbleYes;` because it's in the debug data and I'm not sure what to do with it, but someone else might

**Review by Yanis42 — APPROVED** [review]

**Pull request merged**